clang/test/SemaSYCL/sycl-no-rdc.cpp

```diff
@@ -0,0 +1,25 @@
+// RUN: %clang_cc1 -fsycl-is-device -verify -fsyntax-only -fsycl-allow-func-ptr -fno-gpu-rdc %s
+SYCL_EXTERNAL void syclExternal();
+
+SYCL_EXTERNAL void notSyclExternal() {}
+
+template <typename KernelName, typename KernelType>
+[[clang::sycl_kernel]] void kernel_single_task(const KernelType& kernelFunc) { // #kernelSingleTask
+  kernelFunc();
+}
+
+void kernel() {
+  // expected-error@+1{{seperate compilation unit without relocatable device code}}
+  syclExternal();
+  notSyclExternal();
+  // expected-error@+1{{seperate compilation unit without relocatable device code}}
```

Comment:
Since this error is being generated when the function is accessed and not when it is called, perhaps change the diagnostic to say "cannot use" vs "cannot call"?

Reply:
done, thanks